File: python/src/two_pointers/reverse_string.py
"""Reverse list of characters in place (two pointers)."""


def reverse_string(chars: list[str]) -> None:
    left, right = 0, len(chars) - 1
    while left < right:
        chars[left], chars[right] = chars[right], chars[left]
        left += 1
        right -= 1

File: typescript/src/two_pointers/reverseString.ts
/** Reverses array of single-character strings in place (two pointers). */
export function reverseString(chars: string[]): void {
  let left = 0;
  let right = chars.length - 1;
  while (left < right) {
    const t = chars[left]!;
    chars[left] = chars[right]!;
    chars[right] = t;
    left++;
    right--;
  }
}
